solidity/contracts/bridge/Bridge.sol

```diff
@@ -795,10 +935,534 @@ contract Bridge is Ownable {
+                keccak256(redeemerOutputHash),
+            "Redeemer output hash must not be equal to the wallet PKH"
+        );
+
+        require(
+            amount >= redemptionDustThreshold,
+            "Redemption amount too small"
+        );
+
+        // The redemption key is built on top of wallet public key hash
+        // and redeemer output hash pair. That means there can be only one
+        // request asking for redemption from specific wallet to the given
+        // BTC hash in the same time.
+        uint256 redemptionKey =
+            uint256(
+                keccak256(
+                    abi.encodePacked(walletPubKeyHash, redeemerOutputHash)
+                )
+            );
+
+        require(
+            // slither-disable-next-line incorrect-equality
+            pendingRedemptionRequests[redemptionKey].requestedAt == 0,
+            "Pending request with same redemption key already exists"
+        );
+
+        uint64 treasuryFee = redemptionTreasuryFee;
+        uint64 txMaxFee = redemptionTxMaxFee;
+
+        // The wallet's BTC balance should allow to cover the actual BTC amount
+        // transferred to the redeemer along with its share of the transaction
+        // fee. Those two components always sum up to the request's maximum
+        // redeemable amount which is computed as the difference between the
+        // requested amount and the treasury fee. This is why one should
+        // add the redeemable amount to the total requested value for the given
+        // wallet and assert this total value doesn't exceed the current main
+        // UTXO value  which is actually the same as the current wallet balance.
+        pendingRedemptionRequestsValue[walletPubKeyHash] +=
+            amount -
+            treasuryFee;
+        require(
+            mainUtxo.txOutputValue >=
+                pendingRedemptionRequestsValue[walletPubKeyHash],
+            "Insufficient wallet funds"
+        );
+
+        pendingRedemptionRequests[redemptionKey] = RedemptionRequest(
+            msg.sender,
+            amount,
+            treasuryFee,
+            txMaxFee,
+            /* solhint-disable-next-line not-rely-on-time */
+            uint32(block.timestamp)
+        );
+
+        emit RedemptionRequested(
+            walletPubKeyHash,
+            redeemerOutputHash,
+            msg.sender,
+            amount,
+            treasuryFee,
+            txMaxFee
+        );
+
+        bank.transferBalanceFrom(msg.sender, address(this), amount);
+    }
+
+    /// @notice Used by the wallet to prove the BTC redemption transaction
+    ///         and to make the necessary bookkeeping. Redemption is only
+    ///         accepted if it satisfies SPV proof.
+    ///
+    ///         The function is performing Bank balance updates by burning
+    ///         the total redeemed amount from Bridge balance and transferring
+    ///         the treasury fee sum to the treasury address.
```

Comment:
I think it makes sense to emphasize that the redeemed Bitcoin balance is burned. We are not burning the entire amount that was requested for redemption, a part of it goes to the treasury. We do burn whatever left the ECDSA wallet though. 

```suggestion
    ///         The function is performing Bank balance updates by burning
    ///         the total redeemed Bitcoin amount from Bridge balance and 
    ///         transferring the treasury fee sum to the treasury address.
```

Reply:
See https://github.com/keep-network/tbtc-v2/pull/128/commits/bdcf98ca90b17450b16cfe59d235ce8c119b611b